gremlin-go/driver/graph.go

```diff
@@ -120,3 +121,63 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, fmt.Errorf("path does not contain a label of '%s'", key)
 	}
 }
+
+// SetTemplate describes the necessary methods that need to be implemented to make a user-defined Set implementation.
+// Set is the default implementation that should be used but SetTemplate can be used for a custom implementation if needed.
+type SetTemplate interface {
+	convertSetToSlice() ([]interface{}, error)
+	convertSliceToSet([]interface{}) (SetTemplate, error)
+}
+
+// Set is a default custom declaration since Go does not natively provide this feature.
+type Set struct {
+	Objects []interface{}
+}
+
+func (s *Set) convertSetToSlice() ([]interface{}, error) {
+	return s.Objects, nil
+}
+
+func (s *Set) convertSliceToSet(in []interface{}) (SetTemplate, error) {
+	slice, err := interfaceToSlice(in)
+	if err != nil {
+		return nil, err
+	}
+	s.Objects = slice
+	return s, nil
+}
+
+// Construct a new Set with filtered values
+func newSet(in interface{}) *Set {
+	slice, _ := interfaceToSlice(in)
+	filtered := removeDuplicateValues(slice)
+	s := new(Set)
+	s.Objects = filtered
+	return s
+}
+
+func interfaceToSlice(in interface{}) ([]interface{}, error) {
```

Comment:
So, we dont need this method, but we do need to add a Add() and Remove() method. Remove might be a bit tricky and if it is too finnicky we can neglect it.

Reply:
agree